addons/web/static/tests/fields/basic_fields_tests.js

```diff
@@ -63,7 +63,6 @@ QUnit.module('basic_fields', {
                     qux: 0.44444,
                     p: [],
                     timmy: [],
-                    trululu: 4,
```

Comment:
I'm a bit concerned with those changes in tests. We don't update those data, as it may change test semantics without notice. Adding a field is fine in general, but I do not recommend other changes. Why would you need to do so?

Reply:
This is a legitimate concern. I updated values only when there was an issue for it, and I did investigate why in each case.
Hopefully I can remember each of them and give you proper reasons.

This one is problematic because it attempts to link a record (id 4) that has not been created yet (when creating id 1 which is defined before).

I tried to change the order of creation of records instead of removing the value, but a lot of tests depend on that order, for example to update data in place with index access, so that would be really annoying to update them all.

Then I just tried to remove this annoying value and nothing broke so I kept it removed.

Reply:
You mean because record 4 is defined after record 1? It would be nice if the mock server could handle such cases, as it is more convenient to write. Actually, I'm surprised you this doesn't happen more often.

Reply:
Ok, I will relax the integrity check during init to what it was previously (non existent), as there is no consensus and this doesn't matter to what I'm trying to do with this PR. It was just to make the mock more robust and to more easily spot mistakes during my changes.

Then I'll revert the changes in test that I had to do because of this.